plugin/trino-hive/src/main/java/io/trino/plugin/hive/HivePageSourceProvider.java

```diff
@@ -196,6 +206,12 @@ public ConnectorPageSource createPageSource(
                 ColumnMetadata<OrcType> columnMetadata = orcPageSource.getColumnTypes();
                 int acidRowColumnId = originalFile ? 0 : ACID_ROW_STRUCT_COLUMN_ID;
                 HiveType rowType = fromOrcTypeToHiveType(columnMetadata.get(new OrcColumnId(acidRowColumnId)), columnMetadata);
+
+                long currentSplitNumber = hiveSplit.getSplitNumber();
+                if (currentSplitNumber >= MAX_NUMBER_OF_SPLITS) {
+                    throw new TrinoException(GENERIC_INSUFFICIENT_RESOURCES, format("Number of splits is higher than maximum possible number of splits %d", MAX_NUMBER_OF_SPLITS));
```

Comment:
Provide actual value of `currentSplitNumber` in the exception message as well

Reply:
Not sure if super beneficial. Most of the time it will be == `MAX_NUMBER_OF_SPLITS` as we are generating splits from sequence; and we throw as soon as we cross the boundary.